CountingBloomFilter.hpp

```diff
@@ -271,61 +252,76 @@ template<typename T>
 void
 CountingBloomFilter<T>::readFilter(const std::string& path)
 {
-	FILE* file;
-	if ((file = fopen(path.c_str(), "rb")) == nullptr) {
+	std::ifstream file(path);
+	if (!file.is_open()) {
```

Comment:
```suggestion
	if (file) {
```
Check if the file is good (which you get by converting to `bool`) rather than `is_open`. A file can be open but in a state of failure.